optaplanner-bom/pom.xml

```diff
@@ -0,0 +1,487 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <!-- TODO In step 3, this must extend optaplanner-parent instead and stop duplicating this metadata -->
+    <!-- But for that to work, optaplanner-parent must have an empty dependencyManagement (so no kie-parent parent) -->
+    <groupId>org.kie</groupId>
+    <artifactId>kie-user-bom-parent</artifactId>
+    <version>7.40.0-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.optaplanner</groupId>
+  <artifactId>optaplanner-bom</artifactId>
+  <packaging>pom</packaging>
+  <version>8.0.0-SNAPSHOT</version>
```

Comment:
This file was duplicated from droolsjbpm-build-bootstrap, which has the 7.x version (and cannot host the 8.x version because there is no 8 there).

Advantages of moving it here:
- Once it extends optaplanner-parent, half of the file can be removed (url, scm, issueManagement, etc)
- Less cross-repo PR's.
- Easier maintenance